### PR TITLE
Add basic docs for Metrics api and sdk

### PR DESCRIPTION
**Pull request by cijothomas**
To be expanded with more contents later.

### CODE REVIEW

File: docs/metrics/customizing-the-sdk/README.md
Comment (utpilla) [inline docs/metrics/customizing-the-sdk/README.md:8]: Maybe use Pascal case? MetricReaders, Views, etc.

File: src/OpenTelemetry.Api/README.md
Comment (utpilla) [inline src/OpenTelemetry.Api/README.md:445]: typo: targeting

File: src/OpenTelemetry.Api/README.md
Comment (utpilla) [inline src/OpenTelemetry.Api/README.md:490]: Do we want to mark this static to make it seem like a good practice?

File: docs/metrics/customizing-the-sdk/README.md
Comment (utpilla) [inline docs/metrics/customizing-the-sdk/README.md:16]: configure*

File: docs/metrics/customizing-the-sdk/README.md
Comment (utpilla) [inline docs/metrics/customizing-the-sdk/README.md:17]: include*

File: docs/metrics/customizing-the-sdk/README.md
Comment (utpilla) [inline docs/metrics/customizing-the-sdk/README.md:67]: How about changing this to:

> `AddSource` method can be called multiple times to add more than one meter.

File: docs/metrics/customizing-the-sdk/README.md
Comment (utpilla) [inline docs/metrics/customizing-the-sdk/README.md:68]: Either have "also" or "as well".

File: docs/metrics/customizing-the-sdk/README.md
Comment (utpilla) [inline docs/metrics/customizing-the-sdk/README.md:91]: Add colon after "Note".

File: src/OpenTelemetry.Api/README.md
Comment (pirgeo) [inline src/OpenTelemetry.Api/README.md:439]:
```suggestion
Metrics in OpenTelemetry .NET are a somewhat unique implementation of the
```

File: src/OpenTelemetry.Api/README.md
Comment (pirgeo) [inline src/OpenTelemetry.Api/README.md:440]:
```suggestion
OpenTelemetry project, as the Metrics API is incorporated directly into the
```

File: src/OpenTelemetry.Api/README.md
Comment (pirgeo) [inline src/OpenTelemetry.Api/README.md:467]:
```suggestion
   doing the instrumentation. The `Meter` instance is typically created once and is
```

File: src/OpenTelemetry.Api/README.md
Comment (pirgeo) [inline src/OpenTelemetry.Api/README.md:478]:
```suggestion
    **Note:**
```
As suggested by @utpilla in https://github.com/open-telemetry/opentelemetry-dotnet/pull/2452/files#r722827442

File: src/OpenTelemetry.Api/README.md
Comment (pirgeo) [inline src/OpenTelemetry.Api/README.md:454]: Is this going to be the only way to instrument applications, or is it still possible to include the `OpenTelemetry.Api` package?

Comment (cijothomas) [reply src/OpenTelemetry.Api/README.md:454]: the only way. There are no plans to add shim/wrapper around this in the API package.

Comment (joaopgrassi) [reply src/OpenTelemetry.Api/README.md:454]: So that means that if I want to add metrics to my library, I need to take a dep on the DiagnosticSource package and not OTel? I can see why that, but I feel it would be also good to have a wrapper, not only because it goes against the spec, but other people coming from other languages have to learn, yet this new thing which is not so obvious. 🤔

File: docs/metrics/customizing-the-sdk/README.md
Comment (reyang) [inline docs/metrics/customizing-the-sdk/README.md:66]:
```suggestion
provider. The name of the `Meter` (case-insensitive) must be provided as an argument to
```

File: docs/metrics/customizing-the-sdk/README.md
Comment (reyang) [inline docs/metrics/customizing-the-sdk/README.md:68]:
```suggestion
meters. It also supports wild-card subscription model.
```

File: docs/metrics/customizing-the-sdk/README.md
Comment (reyang) [reply docs/metrics/customizing-the-sdk/README.md:65]: Need to revisit all the changes in this PR to use AddMeter once the name change is merged.

File: docs/metrics/customizing-the-sdk/README.md
Comment (reyang) [inline docs/metrics/customizing-the-sdk/README.md:93]:
```suggestion
`Meter`s to the provider. It is recommended to leverage the wildcard
```

File: src/OpenTelemetry.Api/README.md
Comment (cijothomas) [reply src/OpenTelemetry.Api/README.md:454]: If you take a dependency on Otel.API, then it automatically brings in DS package.
This is the same model we use for tracing (and logging) as well.
We added wrapper for tracing, because the Activity API had lot of naming differences (because Activity existed before OTel came into existence.), and many folks felt the need for a wrapper. 

Metric API in .NET was built ground up, closely following OTel spec, so there are not many differences, warranting the cost of maintaining another wrapper. Having said that, if there is any strong reason to have a wrapper - please create an issue and we can discuss more.

File: src/OpenTelemetry.Api/README.md
Comment (reyang) [inline src/OpenTelemetry.Api/README.md:490]:
```suggestion
    static Counter<long> MyFruitCounter = meter.CreateCounter<long>("MyFruitCounter");
```

**Review by reyang — APPROVED** [review]
LGTM.

**Comment by cijothomas** [comment]
Thanks @joaopgrassi @pirgeo @utpilla for reviews! I have resolved all comments. If I can get another review, that'd be great before merge!

**Review by alanwest — APPROVED** [review]

**Pull request merged**
